conans/test/package_copier_test.py

```diff
@@ -0,0 +1,126 @@
+import unittest
+from conans.test.utils.test_files import temp_folder
+from conans.test.tools import TestBufferConanOutput
+from conans.client.userio import UserIO
+import sys
+from conans.client.package_copier import PackageCopier
+from conans.util.files import mkdir, save, load
+from conans.model.ref import ConanFileReference, PackageReference
+from conans.paths import SimplePaths
+import os
+
+
+class MockedBooleanUserIO(UserIO):
+
+    def __init__(self, answer, ins=sys.stdin, out=None):
+        self.answer = answer
+        UserIO.__init__(self, ins, out)
+
+    def request_boolean(self, msg, default_option=None):
+        self.out.info(msg)
+        return self.answer
+
+
+class PackageCopierTest(unittest.TestCase):
+
+    def testCopier(self):
+        output = TestBufferConanOutput()
+        userio = MockedBooleanUserIO(True, out=output)
+        paths = SimplePaths(temp_folder())
+        copier = PackageCopier(paths, userio)
+
+        # Create some packages to copy
+        reference = ConanFileReference.loads("Hello/0.1@lasote/testing")
+        self._create_conanfile(reference, paths)
+        self._create_package(reference, "0101001", paths)
+        self._create_package(reference, "2222222", paths)
+
+        # Copy all to destination
+        copier.copy(reference, ["0101001", "2222222"], "lasote", "stable", force=False)
+        new_reference = ConanFileReference.loads("Hello/0.1@lasote/stable")
+        self._assert_conanfile_exists(new_reference, paths)
+        self._assert_package_exists(new_reference, "0101001", paths)
+        self._assert_package_exists(new_reference, "2222222", paths)
+        self.assertIn("Copied Hello/0.1@lasote/testing to Hello/0.1@lasote/stable", output)
```

Comment:
I suggest reviewing this output once the scoped output PR is merged, I'd say this output can be improved:
- Copied all
- Hello/0.1@lasote/testing:ID copied to Hello/0.1@user/stable